### PR TITLE
chore: Fix Style recalcs for stickyCanvasArena

### LINKED ISSUE #23777: [Task]: Refactor sticky canvas Arena to avoid unnecessary forced reflow
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

Refactor sticky canvas Arena to avoid unnecessary forced reflow. This is done by enabling the canvas to observe only when required i.e, only when user is using,
- [ ] Drag to select
- [ ] Dragging widgets

### PR DESCRIPTION
**Pull request by rahulramesha**
## Description
Refactor sticky canvas Arena to avoid unnecessary forced reflow. This is done by enabling the canvas to observe only when required i.e, only when user is using,
- Drag to select
- Dragging widgets

Fixes #23777

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video

## Type of change

- Chore (housekeeping or task changes that don't impact user perception)

## How Has This Been Tested?
- Manual

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by marks0351 — APPROVED** [review]

**Review by marks0351 — DISMISSED** [review]

File: app/client/src/pages/common/CanvasArenas/StickyCanvasArena.tsx [hunk 1]
```diff
@@ -111,26 +113,35 @@ export const StickyCanvasArena = forwardRef(
...
     };
+
     const observeSlider = () => {
       interSectionObserver.current.disconnect();
       if (slidingArenaRef && slidingArenaRef.current) {
         interSectionObserver.current.observe(slidingArenaRef.current);
       }
     };
 
+    const observeSliderIfNecessary = () => {
```
Comment (SatishGandham) [inline app/client/src/pages/common/CanvasArenas/StickyCanvasArena.tsx:136]: Why a wrapper function?
Lets just add the condition to the oberveSlider itself. It is also conditional anyway.
Don't forget to add a comment on why we are doing it.

**Review by SatishGandham — CHANGES_REQUESTED** [review]

**Review by SatishGandham — APPROVED** [review]

**Review by marks0351 — DISMISSED** [review]

**Review by SatishGandham — APPROVED** [review]

**Review by SatishGandham — APPROVED** [review]

**Pull request merged**